### ISSUE TITLE
Deprecated function Call.PC() breaks tinygo compatibility?

### ISSUE THREAD

**Issue by dkegel-fastly** [go-stack/stack#31]
Following a rabbit hole of dependencies that starts at https://github.com/tinygo-org/tinygo/issues/2153, I noticed that 

```
$ git clone https://github.com/go-stack/stack
$ cd stack
$ go test
```

succeeds, but

```
$ tinygo test
```

fails with 

```
stack.go:164:17: c.frame.PC undefined (type runtime.Frame has no field or method PC)
```

That's 
```
162 // Deprecated: Use Call.Frame instead.
163 func (c Call) PC() uintptr {
164         return c.frame.PC
165 }
```

So for what it's worth, the deprecated function Call.PC() is causing tinygo build failure.  Commenting it out lets tinygo test pass.

**Comment by ChrisHines** [comment]
Indeed. Tinygo's [`runtime.Frame` type](https://pkg.go.dev/github.com/tinygo-org/tinygo/src/runtime#Frame) is not fully compatible with Go's [`runtime.Frame` type](https://pkg.go.dev/runtime#Frame). It is missing several fields.

The best solution here is to move `func (c Call) PC() uintptr` into a separate file that specifies a `!tinygo` build tag.

Would you like to submit a PR for that?

**Comment by dkegel-fastly** [comment]
I must have been smoking something.  Fixing that one problem uncovers another one now:
```
stack_test.go:580:6: fn.FileLine undefined (type *runtime.Func has no field or method FileLine)
```
so maybe it's best to leave sleeping dogs lie for a bit, until the need for a fix is stronger.

**Comment by ChrisHines** [comment]
Sure. Tinygo's `runtime` package is quite a bit smaller than Go's, so any code that depends on `runtime` is likely to have issues like this.
